### PR TITLE
Add `params_to_tune` for `SARIMAXModel` model

### LINKED ISSUE #1194: Add `params_to_tune` for arima models
### 🚀 Feature Request

Add `params_to_tune` for arima models.

### Proposal

Sources for the grid:
- pycaret
  - [sktime](https://github.com/pycaret/pycaret/blob/master/pycaret/containers/models/time_series.py#L656)
  - [statsforecast](https://github.com/pycaret/pycaret/blob/master/pycaret/containers/models/time_series.py#L748)
- kats
  - [arima](https://github.com/facebookresearch/Kats/blob/992b95e1a8c34402756ffea2376189568de85a2c/kats/utils/parameter_tuning_utils.py#L222)
  - [sarima](https://github.com/facebookresearch/Kats/blob/992b95e1a8c34402756ffea2376189568de85a2c/kats/utils/parameter_tuning_utils.py#L312)

Suggested grid for `SARIMAXModel`:
```
{
    "p": IntUniformDistribution(low=1, high=6, step=1),
    "d": IntUniformDistribution(low=1, high=3, step=1),
    "q": IntUniformDistribution(low=1, high=6, step=1),
    "P": IntUniformDistribution(low=0, high=2, step=1),
    "D": IntUniformDistribution(low=0, high=1, step=1),
    "Q": IntUniformDistribution(low=0, high=1, step=1),
    "trend": CategoricalDistribution(["n", "c", "t", "ct"]),
}
```

Suggested grid for `AutoARIMAModel`:
```
{}
```

To discuss:
- Can we do something with `S` parameter in `SARIMAXModel` (It is the seasonality)? The problem is that it depends on the data. For example, we can just keep the value that was put by the user and don't change it.
- How to generate `order` for `SARIMAXModel` from `p`, `d`, `q` (same for `seasonal_order`)? In `pycaret` they generate all combinations and put them into `CategoricalDistribution`.
  - It will be solved in #1201.
- Do we really need the grid for `AutoARIMAModel`?
  - No, we don't need it.

### Test cases

Look at #1184.

### Additional context

Is blocked by #1201.

### PR DESCRIPTION
**Pull request by Mr-Geekman**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look at #1194.

## Closing issues
Closes #1194.

### CODE REVIEW

File: etna/models/sarimax.py [hunk 1]
```diff
@@ -374,9 +381,9 @@ class _SARIMAXAdapter(_SARIMAXBaseAdapter):
 
     def __init__(
         self,
-        order: Tuple[int, int, int] = (2, 1, 0),
-        seasonal_order: Tuple[int, int, int, int] = (1, 1, 0, 12),
-        trend: Optional[str] = "c",
+        order: Tuple[int, int, int] = (1, 0, 0),
```
Comment (alex-hse-repository) [inline etna/models/sarimax.py:384]: Why did you change the default values?

File: tests/test_models/test_sarimax_model.py [hunk 2]
```diff
@@ -169,7 +170,7 @@ def test_decomposition_hamiltonian_repr_error(dfs_w_exog, components_method_name
 )
 @pytest.mark.parametrize("trend", (None, "t"))
 def test_components_names(dfs_w_exog, regressors, regressors_components, trend, components_method_name, in_sample):
-    expected_components = regressors_components + ["target_component_sarima"]
+    expected_components = regressors_components + ["target_component_arima"]
```
Comment (alex-hse-repository) [inline tests/test_models/test_sarimax_model.py:173]: ???

**Review by alex-hse-repository — CHANGES_REQUESTED** [review]

File: etna/models/sarimax.py (hunk 1 above)
Comment (Mr-Geekman) [reply etna/models/sarimax.py:384]: Because our values in a library doesn't correspond to the values in statsmodels library. There were two problems:
1. Our documentations contradicts the default values we set. For example, we write in documentation that by default we use AR(1) model, but we use ARIMA(2, 1, 0) model.
2. Our default values are rather strange. For what reason do we set `seasonal_order=(x, x, x, 12)`? Why do we set seasonality of 12?

File: tests/test_models/test_sarimax_model.py (hunk 2 above)
Comment (Mr-Geekman) [reply tests/test_models/test_sarimax_model.py:173]: With new default values we by default doesn't have seasonality (because the last component of `seasonal_order` is equal to 0). And in this case we name component `arima` instead of `sarima`.

File: etna/models/sarimax.py (hunk 1 above)
Comment (alex-hse-repository) [reply etna/models/sarimax.py:384]: No ideas))

**Review by alex-hse-repository — APPROVED** [review]
